doc/utils/manifest.md

```diff
@@ -38,6 +48,28 @@ This section and all its fields are required. It contains metadata helping to bu
 | `build_directory` | Path of the build directory (i.e the directory where the `Makefile` or `Cargo.toml` file can be found) |
 | `devices`         | The list of devices on which the application can be built.                                             |
 
+#### `[use_cases]`
+
+This section is optional. It contains metadata helping select build options depending on use cases
+The VSCode extension leverages this section to provide alternative build targets.
+
+| Field name   | Description                                                             |
+|--------------|-------------------------------------------------------------------------|
+| `<use_case>` | Options string: <ul><li>Environment variables definitions for C applications (e.g. `DEBUG=1`)</li><li>Valid Cargo build options for Rust applications (e.g. `--outdir mydir`)</li></ul> |
+
+This specifies that in order to build for `<use_case>`, the options string must be provided in the build command line.
+You are free to add any use case you wish to have a VSCode build target for.
+> [!WARNING]
+> The use case `"default"` refers to a standard build with no option. It is implicit and can't be redefined.
+
+Example:
+```
+[use_cases] # Coherent build options that make sense for your application
+debug = "DEBUG=1"
+test = "TESTING=1 TEST_PUBLIC_KEY=1"
+my_variant = "COIN=MY_VARIANT"
```

Comment:
I'm wondering if variant shouldn't be handled differently?
Cause we might need it for other tools, like CI tests.
Also it will probably need a lot of use_case combination (like if I want my variant + debug ?) How is it going to look in vsCode? e.g. can i choose multiple use_case at the same time?

Reply:
I suggest seeing if this proposal fits the need, then if needed handle the variant in step 2
Most apps have few variants (less than 3), in this case it looks like:
```
foo = ...
foo_debug = ...
bar = ...
bar_debug = ...
baz = ...
baz_debug = ...
```
6 lines is very acceptable

Extracting the variant from the makefile to the manifest is easy, but then having the makefile read the manifest for the info in a clear way is difficult

Reply:
I was more suggesting having the variant list in both the makefile and the manifest, rather than removing it from the makefile which indeed would be tricky.

Reply:
I definitively think that the variant section would help improve our CI.

Reply:
Note that variants might not be available on all devices...

Reply:
Maybe we can define a variant section with shared variant and device specific variant?